edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/DeviceScopeController.cs

```diff
@@ -155,6 +155,33 @@ async Task<EdgeHubScopeResult> HandleGetDeviceAndModuleOnBehalfOfAsync(string ac
             return result;
         }
 
+        bool NeedToRefreshIdentity(Option<ServiceIdentity> identityOption)
+        {
+            // Default refresh to true if we don't have the identity yet.
+            bool needToRefresh = true;
+
+            identityOption.ForEach(id =>
+            {
+                if (id.ModuleId.Contains(Constants.EdgeAgentModuleId) ||
```

Comment:
Do we need to special case EH/EA? We can prob do the same for any identity that is trying to connect, but has Auth type None, right?